### DISCUSSION TITLE
Platform UI: Feedback

### DISCUSSION THREAD

**Discussion by matmair** [inventree/InvenTree#5328] [Ideas]
> [!IMPORTANT]
> Platform UI is the only UI starting with 1.0 - the last version with CUI is 0.17.x
> If you want to submit feedback regarding PUI: Please do so with a new discussion or open a feature request.

Provide your feedback for [Platform UI](https://demo.inventree.org/platform/), our new frontend, here!

There are more details on why we are doing this [on the blog](https://inventree.org/blog/2023/08/28/react). This is an ongoing effort, there is a lot to do - head over to #5212 to get an overview of what is done and what is already identified as a work item.

If you have suggestions for missing / not ported-over features, you can also mention them on #5212 after checking on the open issues there to make sure they are not already open item.

**Comment by rohitsingal1** [comment]
Started using inventree a month back.
Not being a highly technical guy. used the bare metal installation.
So far I have been really happy with the software and the community support behind it.
Any queries or doubts raised by me on the discussion forums have been answered. 
@matmair @SchrodingersGat and others have been really very patient in helping me out.
My main intention of using this software was to check my stock levels and the purchase order requests, which are easily maintained in inventree.
Still there is a long way to go and I wish the entire team behind inventree all the best.
Edit: Sorry this is a generic feedback. Had seen the UI part later. If you find this is not a part of the feedback kindly delete.

**Reply by matmair** [reply]
Thank you for the feedback, I have made the text more indicative of the intended area. I need the link to this topic for UI inclusion ;-)

**Comment by sintech** [comment]
@matmair What login/password we should use? admin/inventree doesn't work.

**Reply by wolflu05** [reply]
Seems like the server selector is broken if nothing is stored in your browser. You need to first select the correct server. @matmair you see, we talked about that, that definitely needs to be set correctly by default for every deployment based on Django settings somehow.

**Reply by sintech** [reply]
I have managed to login using server "tMMrz4", but it looks like this
<details><summary>Details</summary>

![изображение](https://github.com/inventree/InvenTree/assets/10057327/567459a5-2c03-47ec-9655-24bd174d3e15)

</details>

**Reply by matmair** [reply]
The default selection points to localhost, the PR fixing that is in the same PR that contains the Feedback section that I need the link for; I did not think that suddenly people would start reacting to discussions as we got only 3 responses in a week last time. @wolflu05 I am gladly reviewing your patches.

**Reply by matmair** [reply]
@sintech yes the start page should be ready for review in the next few hours

**Comment by Century1026** [comment]
i can't find edit parameter template in new UI

**Reply by SchrodingersGat** [reply]
@Century1026 it is in the new "admin center":

https://demo.inventree.org/platform/settings/admin/part-parameters

**Reply by Century1026** [reply]
i saw it, thank you so much!

**Reply by matmair** [reply]
where did you look in the new UI? Should we add an information there?

**Reply by Century1026** [reply]
@matmair 

you mean the how i managed to find the "edit parameter template"? i first check all the items in the "system settings" of the new UI, as all the settings are included in a single page in the old UI, so I didn't check other pages.
then I searched source code and found that there existed string such as "add parameter template" and "edit parameter template". i thought i might missed it, i even went back to checke part/category/index/parameters. apparently it doesn't contain such button, and the old UI neither.
finally i saw the "provide feedback" link in the home page of the new UI, and i am here.

i think the logic here in the new UI is that the settings with switches are in one group while settings with customized tables are in another group which is arranged clearer and more user-friendly than the original one.
so, yeah, with the perfect settings arrangement, it's necessary to add information to notify that some settings have moved to other pages.
since there's an "add" button on the top left in pages like "/part/category/index/parts" and "/part/category/index/subcategories", users like me would subconsciously wanna "add" part parameter in "/part/category/index/parameters". therefore, you could consider adding a floating hint on the top left of the page for the first time user opening it.

**Reply by matmair** [reply]
Thank you for the feedback, that helps a lot! I think the hints are a good suggestion. I will look into dismisable hints / tips in general.

Would you find a search widget on top of the settings tabs helpful? We could maybe hint to the admin center from there and vice versa.

**Reply by Century1026** [reply]
@matmair 

In my view, adding a search widget is to help user save their time on navigation.
If your settings is too large to hold within a page and has multiple sub items, adding a search widget will reduce the time directing to  items.
Since the settings in new UI are divided into two organized groups and there aren't many items in each group, it's super easy to navigate to any items. Therefore, it's unnecessary to add a search widget.

By the way, i forgot to mention that i also check the [document](https://docs.inventree.org/en/latest/part/parameter/#create-template) to double-check that the feature exists. Therefore, updating the document seems to be the easiest way to hint users.

Search widgit is cool but if users don't know the feature exist, they wouldn't even search for it.

I really love the look of the new UI. Keep the work going and good luck!

**Comment by Century1026** [comment]
@matmair This page is not available. 

> There are more details on why we are doing this [on the blog](https://inventree.org/blog/2023/07/27/react).

**Reply by SchrodingersGat** [reply]
Here it is - https://inventree.org/blog/2023/08/28/react

**Reply by Century1026** [reply]
Thanks, it makes a lot of sense why devevloping PUI.

However, I'm confused two features in classic UI are not avaliable in PUI which are useful.
1. detailed information of "Recently Updated" etc.
In PUI, they're replaced with numbers only, which is less intuitive.
For example, when a stock item is expired, I can quickly locate it in this section, rather than heading stock items list and filter it out.
![image](https://github.com/inventree/InvenTree/assets/106154819/12582bf9-2f79-46ee-b0dd-7ef0687e4781)
2. tree view of stock location
With tree view, we can inspect stock quantity of each location in an organized and intuitive way.
![image](https://github.com/inventree/InvenTree/assets/106154819/bc556af1-c7ee-408b-9a4b-451b93084c1b)

I hope these features be brought back, and I'm willing to contribute to it if your team think it's necessary.

**Reply by SchrodingersGat** [reply]
1. dashboard

The new dashboard is currently only a placeholder and will require some further work. While we won't be fully reproducing the CUI dashboard, certainly there is scope for retaining some of the functionality.

2. tree view

It is likely that we could support a "tree" structure in the stock location (and part category) tables but would require some extra work at this stage.

**Comment by Valdoise** [comment]
InvenTree version 0.15.1 | 196

The new interface looks great ! Apart from the features that will be implemented in #5212, the only feature I am missing is the ability to open a new tab with a CTRL+click (or middle click) on the UI elements. When I am browsing for parts, I like to go through the list and open every part in a new tab. I also like to be able to open new tabs with the navbar elements (parts, stock, ...).

EDIT : looks like CTRL+click is working on part list, but not on navbar. Middle click never works.

**Reply by SchrodingersGat** [reply]
@Valdoise thanks for the feedback! I've opened a new issue about this here - https://github.com/inventree/InvenTree/issues/7271

The "middle click" button might be tricky to get working - I have not been able to work it out so far.

**Comment by pyhtoncoder0** [comment]
I also can't seem to get logged into the demo to report issues. However the main issue I see in the UI is you can't change your camera source for the QR scanning. On the stable version if your local client has 2 cameras it allows you to choose what cam to use. On the new UI it forces you to use the webcam only. Even if the machine has a second camera on the back. In the case of surface and tablet style PCs or maybe a USB webcam this would be an issue. Otherwise it looks great. I love this project so much. You guys do indeed do a good job on it.

**Reply by matmair** [reply]
The improved scanning interface has selection, please try if that works https://demo.inventree.org/platform/scan

**Comment by BryanJacobs** [comment]
I expected to be able to edit single fields on a part or stock item by clicking on the individual field and having the label converted to a text input, instead of needing to go to a separate edit screen. That's the usual data entry pattern in modern web apps.

But the platform UI still requires clicking a meatball menu, waiting, clicking "edit part", waiting, editing the field, clicking save, and waiting a third time.

**Reply by shapirus** [reply]
Absolutely, these days you almost intuitively click on a value and expect it to turn into an input field. This would improve convenience a lot, and it's actually something that should be reasonably easy to do in the new UI.

**Reply by matmair** [reply]
We are happy to review PRs. The current focus is not on changing the general usage mechanisms, we are trying to port everything over first.

**Comment by pyhtoncoder0** [comment]
I can’t figure out how to log in to this link. I forgot all about this tbh.
I can poke around the docs again later today.

pyhton-coder

**Reply by SchrodingersGat** [reply]
https://inventree.org/demo

**Comment by shapirus** [comment]
Any chance it can be made possible to see part parameters and stock counts on the same page?

Use case: I need to pick a part matching a certain range of values that a design allows. I open "part parameters" in a respective category, see a few parts that fall into the acceptable range of values, but how many of each of them do I have? No easy way to tell: need to open each part in a new tab, switch to that tab, see the stock count, close the tab.

It would be so much better if there simply was an additional "Stock" column showing total number of each part available. The "Parts" section has it, but it doesn't have parameters.

**Comment by lokyi792** [comment]
Is it possible to revert to the old layout?
Just did a new install with the latest build, can't change Default currency (goes to a blank page) and probably a few other things I haven't found yet.

**Reply by matmair** [reply]
The old UI is still shipped but has a limited remaining time so it would be good if you raised bug reports for the issues you find.

**Reply by lokyi792** [reply]
Roger, I just realised you could just knock the URL back to root and it's back to the old one.
Ok, couple of things; Importing parts etc. Can't see anywhere in the new menu to do that.
We'd literally just set up a bunch of suppliers/parts/stock in v14, decided to move to a Docker build and use v15... and now trying to export/import it in to the new system...

**Comment by lokyi792** [comment]
15.4
Template/Report editor is borked.
I'm playing around trying to learn to modify the Purchase Order reports.
I had first one setup in there as a direct copy from the github example unedited.
I created a second copy where I removed the product image from the line item.
I created a third one to start playing with formatting and adding new elements.

Once I started playing around in the third one, as soon as you hit save and reload preview it overwrites the other two with the same code.
I'm guessing this is not intended behavior, else why would you have the option for multiple reports?

**Reply by SchrodingersGat** [reply]
@lokyi792 definitely not intended behaviour. Can you please raise a new issue with some screenshots / screen capture to demonstrate the issue?

**Reply by lokyi792** [reply]
> @lokyi792 definitely not intended behaviour. Can you please raise a new issue with some screenshots / screen capture to demonstrate the issue?

Also just noticed when I deleted the second two, the contents of the first one was gone as well...

Will give my brain a break from code and have a go at repeating and screenshotting later. It's definitely repeatable.

**Reply by SchrodingersGat** [reply]
Thanks, please do. It sounds serious!

**Reply by lokyi792** [reply]
Just an addendum to this, we're going to spin up a second instance, I found using one for both out departments is going to get messy real quick.
I was hoping (and maybe a future feature) that parts & stock could be actually filtered by location to an owner (user or group) rather than just seeing the whole lot regardless.
Will test this on the new instance as well jsut to make sure it's not just an issue with the first one.

**Comment by martonmiklos** [comment]
Hi Folks,

I would have a development related question: is there any planned milestone from where you will only add features to the new UI?

**Reply by matmair** [reply]
We are already past that

**Comment by mpigerl** [comment]
First of all, I really want to thank the great developers of InvenTree. Really great work. We are using it for four month now and we are super happy that this project exists. Now to my question / challenge. We are still using the old UI, because I cannot log in with the new UI. 
If I switch to the new login screen, the login screen opens, but after entering the correct credentials, I get the error message "login failed - check input and try again". The old GUI works without any problems. Inventree runs at Version 0.15.7 (but same problem with older versions). Installation is a production installation using docker. API 196, Python Version 3.11.9, Django 4.2.14. 
Are there any ideas, where I should start searching the problems?

**Reply by wolflu05** [reply]
Hi, please post a Screenshot of the browser devtools console and network tab after you try logging in (feel free to redact sensitive information like your domain). And make sure to open the devtools before hitting login, via ctrl + shift + I.

**Reply by mpigerl** [reply]
Thx for your answer 

<img width="1506" alt="Bildschirmfoto 2024-08-01 um 13 23 12" src="https://github.com/user-attachments/assets/81879975-58d4-4388-9f94-8411835ec2a0">

**Reply by wolflu05** [reply]
Are you sure, that's your username? I don't think email works too, in case that is your email. So please try with your username.

**Reply by mpigerl** [reply]
yes, our usernames are equal to email. We did it because of SSO with MS as identify provider. But I tried it here as a  local login, because I do not want to make things more complicate. The username (which looks like an email) works with the old UI. 

<img width="987" alt="Bildschirmfoto 2024-08-02 um 09 35 39" src="https://github.com/user-attachments/assets/0a04a49c-af74-4975-88be-404c6478b624">

**Reply by wolflu05** [reply]
Ok, then it would be helpful if you click on the url on the right side of a red message where it says 401 in your previous screenshot. This should bring you to the network tab of the browser devtools. Then select response in the appearing panel and post a screenshot of it.

**Comment by mpigerl** [comment]
<img width="1490" alt="Bildschirmfoto 2024-08-02 um 10 23 08" src="https://github.com/user-attachments/assets/f1506b1d-40ab-4e75-942b-adcd8e3d10ec">
<img width="1489" alt="Bildschirmfoto 2024-08-02 um 10 22 52" src="https://github.com/user-attachments/assets/da906570-a9d4-4127-94f6-3a45298e5e4f">
<img width="1374" alt="Bildschirmfoto 2024-08-02 um 10 20 49" src="https://github.com/user-attachments/assets/233854a0-7e8c-42cf-b64a-9feb47c9e7b7">

MFA is correct, but no screen is displayed to enter the saftey token from the authentificator app, as the old UI does. However credentials are provided correctly. Could be a reverse proxy configuration issue?

**Reply by wolflu05** [reply]
Oh ok, MFA is currently not supported by 0.15 (stable). But this was implemented in #7469 which will get released as part of 0.16

As a workaround, have you already tried to login through CUI and then just navigate to /platform. Normally the session should be picked up and you can use PUI too.

**Comment by mpigerl** [comment]
Thank you very much for this fast and great answer

**Comment by nall** [comment]
version 0.16.9.

I can't find any way to create location types in the PUI. In CUI there was a "Add Location Type" button when creating a new Stock Location, but this button isn't present in PUI.

**Reply by nall** [reply]
And as these things go, I immediately found it 🤦‍♂️ 

Going to the Admin Center there is a section for Location Types.

**Comment by chadleylyell** [comment]
Hello, can you guys please implement a web based admin control to change the icon and favicon? If not, will you please explain why not?

**Reply by SchrodingersGat** [reply]
@chadleylyell you are welcome to submit a PR to implement this!

**Reply by chadleylyell** [reply]
@SchrodingersGat A "PR"?

**Reply by matmair** [reply]
A pull request; GitHub lingo for a merge request; A patch that changes code - in this case implementing the functionality you are inquiring about

**Reply by chadleylyell** [reply]
What if this is something I don't know how to code? I didn't think this was what I was going to have to do in order to have this implemented.

**Reply by SchrodingersGat** [reply]
@chadleylyell you are under no obligation to implement this feature, but by the same token, neither are we, as unpaid developers who are writing this off our own back in our "free time" :)

I would suggest that if you want to see custom icon support, the first step is to [open a new issue](https://github.com/inventree/InvenTree/issues/new?template=feature_request.yaml) so that we can track this feature request.

Please note that this project receives no direct funding and we have a huge backlog of un-funded feature requests from users who (through no fault of their own) are unable to contribute the code themselves. Perhaps you would be willing to fund the development of this instead?

**Reply by chadleylyell** [reply]
@SchrodingersGat That is something I truly understand! Been working on a small electronic project for about a year and a half now. I definitely did not intend that message in a negative way whatsoever, I actually was chilling at lunch when I replied, haha. What would be needed to fund such a implementation, like pricewise? :)

**Comment by ondras12345** [comment]
Currently (0.17.1), PUI is way less readable in narrow windows than CUI:
<details><summary>CUI</summary>
<img src="https://github.com/user-attachments/assets/f58096c7-6555-413b-ac05-6cc9093e24a7">
</details>
<details><summary>PUI</summary>
<img src="https://github.com/user-attachments/assets/ff0b81c5-cdb0-484d-a3e4-c59a60bf31c5">
</details>

This isn't extremely narrow, either: it is a full screen window on a 1080x1920 (vertical) monitor.

**Reply by SchrodingersGat** [reply]
@ondras12345 thanks, that's pretty gross! I'll start looking into this. Any inputs for better responsive design here would be great, especially from anyone with UX / react experience.

**Reply by SchrodingersGat** [reply]
In particular the vertical "squashing" of the text (when the window becomes quite narrow) is visually unappealing!

**Reply by wolflu05** [reply]
I think the boxes should just break and be vertically stacked.

**Reply by SchrodingersGat** [reply]
@wolflu05 how would you implement this?

**Reply by wolflu05** [reply]
I think your other PR covers this, haven't seen this when I initially commented here.

**Reply by ondras12345** [reply]
If I understand correctly, you suggested to stack the four boxes (name, stock, active, creation date) vertically.
I seems like that would waste a lot of screen space, and force the user to scroll more. It will be necessary at still lower widths (e.g. mobile), but at this width I'd like to see something like this:
<details>
<img src="https://github.com/user-attachments/assets/7170e5da-06f2-46d7-bb09-653e3ce717e5">
</details>

https://stackoverflow.com/questions/17159708/wrap-table-row-to-the-next-line

Alternatively, the image could be moved to the top to leave more width to the box with name, description, ...

However, I have very little experience with UI design, so please take this with a grain of salt.

**Reply by sintech** [reply]
Just a humble question: do we really need this table near the part image? 
Its contents is already on the screen: category, part name and description.
